src/classes/WalletManager/index.ts

```diff
@@ -76,20 +89,25 @@ export class WalletManager {
      * @returns {Promise<void>}
      */
     public async connect(
-        walletApplication: WalletApplication,
+        walletConnectorId: string,
         walletErrorHandlers?: ConnectWalletErrorHandlers,
     ): Promise<void> {
-        const provider = await detectEthereumProvider<MetaMaskEthereumProvider>()
+        const walletConnector = this._connectors[walletConnectorId]
+
+        if (!walletConnector)
+            throw new Error(EthError.PROVIDER_UNAVAILABLE)
+
+        const provider = await walletConnector.getProvider()
 
         if (!provider) {
             if (walletErrorHandlers?.onProviderUnavailable)
-                return walletErrorHandlers.onProviderUnavailable(walletApplication)
+                return walletErrorHandlers.onProviderUnavailable(walletConnector.id)
 
             throw new Error(EthError.PROVIDER_UNAVAILABLE)
         }
 
         try {
-            this._currentWalletApplication = walletApplication
+            this._currentWalletConnectorId = walletConnector.id
             this._nativeProvider = provider
             this._wrappedProvider = new BrowserProvider(this._nativeProvider as Eip1193Provider, 'any')
```

Comment:
i'm not sure about this, but maybe the connector should return a `new BrowserProvider()` instance and other connector types, like walletconnect will have a different instance type, I don't know, we need to check

Reply:
I don't think it's necessary plus I believe that it is better to have just one place dealing with the `BrowserProvider` instance rather than every connector. The connector should only be responsible for returning an ethereum provider and the `BrowserProvider` wrapping is mainly for the wallet manager logic...